vault/identity_store_util.go

```diff
@@ -333,22 +333,37 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 			// into merging
 			fallthrough
 		default:
+			if !persist {
+				i.logger.Warn(errDuplicateIdentityName.Error(), "alias_name", alias.Name, "mount_accessor", alias.MountAccessor, "entity_name", entity.Name, "action", "delete one of the duplicate aliases")
+				if !i.disableLowerCasedNames {
+					return errDuplicateIdentityName
+				}
+
+				// At this point, identity store is operating case-sensitively.
+				// Persisting is allowed only on the primary.
+				if i.core.ReplicationState().HasState(consts.ReplicationPerformanceSecondary) || i.core.ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
```

Comment:
As I found out today, this won't do what you expect. Perf standby only appears in this call if you're going through a system view. You need to check c.perfStandby.

Reply:
Thanks a bunch! Fixed.